src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenOperators.cs

```diff
@@ -5349,5 +5349,296 @@ static void Main()
+  IL_000c:  br.s       IL_0015
+  IL_000e:  ldloca.s   V_0
+  IL_0010:  call       ""int int?.GetValueOrDefault()""
+  IL_0015:  newobj     ""int?..ctor(int)""
+  IL_001a:  ret
+}");
+            comp.VerifyIL("Program.CoalesceWithNonDefault2", @"{
+  // Code size       15 (0xf)
+  .maxstack  1
+  .locals init (int? V_0)
+  IL_0000:  ldarg.0
+  IL_0001:  stloc.0
+  IL_0002:  ldloca.s   V_0
+  IL_0004:  call       ""bool int?.HasValue.get""
+  IL_0009:  brtrue.s   IL_000d
+  IL_000b:  ldarg.1
+  IL_000c:  ret
+  IL_000d:  ldloc.0
+  IL_000e:  ret
+}");
+            comp.VerifyIL("Program.CoalesceWithNonDefault3", @"{
+  // Code size       23 (0x17)
+  .maxstack  1
+  .locals init (int? V_0,
+                int? V_1)
+  IL_0000:  ldarg.0
+  IL_0001:  stloc.0
+  IL_0002:  ldloca.s   V_0
+  IL_0004:  call       ""bool int?.HasValue.get""
+  IL_0009:  brtrue.s   IL_0015
+  IL_000b:  ldloca.s   V_1
+  IL_000d:  initobj    ""int?""
+  IL_0013:  ldloc.1
+  IL_0014:  ret
+  IL_0015:  ldloc.0
+  IL_0016:  ret
+}");
+        }
+
+        [Fact]
+        public void TestNullCoalesce_NonNullableWithDefault_NoOptimization()
+        {
+            var source = @"
+class Program
+{
+    static string CoalesceNonNullableWithDefault(string x)
+    {
+        return x ?? default(string);
+    }
+
+    static void Main()
+    {
+        void WriteLine(object value) => System.Console.WriteLine(value?.ToString() ?? ""*null*"");
+
+        WriteLine(CoalesceNonNullableWithDefault(""value""));
+        WriteLine(CoalesceNonNullableWithDefault(null));
+    }
+}";
+            var expectedOutput =
+@"value
+*null*";
+            var comp = CompileAndVerify(source, expectedOutput: expectedOutput);
+            comp.VerifyIL("Program.CoalesceNonNullableWithDefault", @"{
+  // Code size        7 (0x7)
+  .maxstack  2
+  IL_0000:  ldarg.0
+  IL_0001:  dup
+  IL_0002:  brtrue.s   IL_0006
+  IL_0004:  pop
+  IL_0005:  ldnull
+  IL_0006:  ret
+}");
+        }
+
+        [Fact]
+        public void TestNullCoalesce_NullableDefault_MissingGetValueOrDefault()
+        {
+            var source = @"
+class Program
+{
+    static int Coalesce(int? x)
+    {
+        return x ?? 0;
+    }
+}";
+            var comp = CreateCompilation(source);
+            comp.MakeMemberMissing(SpecialMember.System_Nullable_T_GetValueOrDefault);
+            
+            comp.VerifyEmitDiagnostics(
+                // (6,16): error CS0656: Missing compiler required member 'System.Nullable`1.GetValueOrDefault'
+                //         return x ?? 0;
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "x").WithArguments("System.Nullable`1", "GetValueOrDefault").WithLocation(6, 16),
+                // (6,16): error CS0656: Missing compiler required member 'System.Nullable`1.GetValueOrDefault'
+                //         return x ?? 0;
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "x").WithArguments("System.Nullable`1", "GetValueOrDefault").WithLocation(6, 16)
```

Comment:
>ERR_MissingPredefinedMember [](start = 37, length = 27)

@AlekseyTs, would it be worth it to try and avoid duplicate diagnostic here, in your opinion?

Reply:
> would it be worth it to try and avoid duplicate diagnostic here, in your opinion?

I don't think this is a unique situation. Given that this is an edge case, I would not worry about the duplication.